### PR TITLE
set xds user agent name everywhere where grpc_csharp_ext is built

### PR DESCRIPTION
**Pull request by jtattermusch**
Followup for https://github.com/grpc/grpc/pull/26189.

Even though this approach works (and I think I've found all the places where grpc_csharp_ext gets built), I don't like it much because it's making the build more complicated and it's very fragile because it's quite easy to accidentally not set the option, especially because there's quite a few locations where it needs to be set.
(at which point C# might end up with wrong XDS client name).

Are there any alternative solutions? (e.g. the C# library sets the xds client name and version at runtime right after it loads the grpc C core library - that way there would be just one location in the code where we need to set the xds user agent suffix).

Update:
we discussed offline and decided that the approach used in this PR is acceptable.

### CODE REVIEW

**Comment by markdroth** [comment]
I've just merged #26189.  Let's try to get this merged before the branch cut for the next release (next week).  Thanks!

File: CMakeLists.txt [hunk 1]
```diff
@@ -248,6 +248,10 @@ else()
   set(_gRPC_CORE_NOSTDCXX_FLAGS "")
 endif()
 
+if (gRPC_XDS_USER_AGENT_IS_CSHARP)
+  add_definitions("-DGRPC_XDS_USER_AGENT_NAME_SUFFIX=csharp" "-DGRPC_XDS_USER_AGENT_VERSION_SUFFIX=2.39.0-dev")
```
Comment (markdroth) [inline CMakeLists.txt:252]: The values of these macros need to include the quotes.

**Comment by jtattermusch** [comment]
@markdroth  this is now ready for final review.

**Comment by jtattermusch** [comment]
the interop test "cancel_after_first_response" looks like a flake. Merging.

**Pull request merged**